fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({
+        todos: data
+      }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    /**
+     * Use fetch to send a POST request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * add the created todo to the state array.
+     *
+     * TIP: Be sure to SERIALIZE the todo in the body with JSON.stringify()
+     * And specify the "Content-Type" header as "application/json"
+     */
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(data => {
+        const newToDos = [...this.state.todos];
+        newToDos.push(data);
+        this.setState({
+          todos: newToDos
+        })
+      })
+
+  }
+
+  toggleCompleted(todoId) {
+    /**
+     * Find the index of the matching todo in the state array.
+     * Find its "isCompleted" status.
+     * Make a new Object containing the opposite "isCompleted" status.
+     * Use fetch to send a PATCH request to `/api/todos/${todoId}`
+     * Then 😉, once the response JSON is received and parsed,
+     * replace the old todo in state.
+     *
+     * TIP: Be sure to SERIALIZE the updates in the body with JSON.stringify()
+     * And specify the "Content-Type" header as "application/json"
+     */
+    const matching = (todo) => todo.id === todoId;
+    const todo = this.state.todos.findIndex(matching);
```

Comment:
`.findIndex()` returns a number, an index so `todo` is a number. You should use `.find()` here, that will return the todo object.